haskell/compile.bzl

```diff
@@ -265,18 +263,13 @@ def target_metadata(
         ghc_args.add("-hide-all-packages")
         ghc_args.add(package_flag, "base")
 
-        package_dbs = ctx.actions.tset(
-            HaskellPackageDbTSet,
-            children = [package_db[name] for name in toolchain_libs if name in package_db]
-        )
-
         ghc_args.add(cmd_args(toolchain_libs, prepend=package_flag))
         ghc_args.add(cmd_args(packages_info.exposed_package_args))
         ghc_args.add(cmd_args(packages_info.packagedb_args, prepend = "-package-db"))
-        ghc_args.add(cmd_args(package_dbs.project_as_args("package_db"), prepend="-package-db"))
         ghc_args.add(ctx.attrs.compiler_flags)
 
         md_args = cmd_args(md_gen)
+        md_args.add(packages_info.nix_packages)
```

Comment:
Is this really Nix specific, or would the same approach be needed for other forms of GHC toolchains? I'm having the goal to upstream to facebook/buck2-prelude in mind. If there's a more general pattern, then it would be good to call that out. (Not necessarily in this PR though, but good to keep in mind as a follow-up)